File: aml.client/src/app/_DTOs/RegisterRequest.ts
import { AccountType } from "../_classes/AccountType";
import { Branch } from "../_models/Branch";

export class RegisterRequest {
  public email: string | undefined;
  public password: string | undefined;
  public accountType: AccountType = 1;
  public name: string | undefined;
  public address: string | undefined;
  public phone: string | undefined;
  public branch: Branch | undefined;
}

File: AML.Server/DTOs/RegisterRequest.cs
using AML.Server.Models;

namespace AML.Server.DTOs
{
    public class RegisterRequest
    {
        public string Email { get; set; }
        public string Password { get; set; }
        public AccountType AccountType { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public Branch Branch { get; set; }
    }
}
